src/parser/Parser.ts

```diff
@@ -945,7 +1082,7 @@ export class Parser {
 
                     function key() {
                         let k;
-                        if (check(Lexeme.Identifier)) {
+                        if (check(Lexeme.Identifier, ...allowedProperties)) {
```

Comment:
This is the new section that you suggested for supporting many reserved words as property names